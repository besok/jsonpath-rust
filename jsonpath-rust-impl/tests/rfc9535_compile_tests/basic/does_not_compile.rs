// Test case: 01_no_leading_whitespace(DISABLED: due to macro limitations)
// Tags: whitespace
// fn test_01_no_leading_whitespace() {
//     ::jsonpath_rust_impl::json_query!( $);
// }

// Test case: 02_no_trailing_whitespace(DISABLED: due to macro limitations)
// Tags: whitespace
// fn test_02_no_trailing_whitespace() {
//     ::jsonpath_rust_impl::json_query!($ );
// }

// Test case: 06_name_shorthand_symbol
// Tags: No tags
fn test_06_name_shorthand_symbol() {
    ::jsonpath_rust_impl::json_query!($.&);
}

// Test case: 07_name_shorthand_number
// Tags: No tags
fn test_07_name_shorthand_number() {
    ::jsonpath_rust_impl::json_query!($.1);
}

// Test case: 16_multiple_selectors_space_instead_of_comma
// Tags: whitespace
fn test_16_multiple_selectors_space_instead_of_comma() {
    ::jsonpath_rust_impl::json_query!($[0 2]);
}

// Test case: 17_selector_leading_comma
// Tags: No tags
fn test_17_selector_leading_comma() {
    ::jsonpath_rust_impl::json_query!($[,0]);
}

// Test case: 18_selector_trailing_comma
// Tags: No tags
fn test_18_selector_trailing_comma() {
    ::jsonpath_rust_impl::json_query!($[0,]);
}

// Test case: 28_empty_segment
// Tags: No tags
fn test_28_empty_segment() {
    ::jsonpath_rust_impl::json_query!($[]);
}

// Test case: 39_bald_descendant_segment
// Tags: No tags
fn test_39_bald_descendant_segment() {
    ::jsonpath_rust_impl::json_query!($..);
}

// Test case: 40_current_node_identifier_without_filter_selector
// Tags: No tags
fn test_40_current_node_identifier_without_filter_selector() {
    ::jsonpath_rust_impl::json_query!($[@.a]);
}

// Test case: 41_root_node_identifier_in_brackets_without_filter_selector
// Tags: No tags
fn test_41_root_node_identifier_in_brackets_without_filter_selector() {
    ::jsonpath_rust_impl::json_query!($[$.a]);
}

fn main() {}
