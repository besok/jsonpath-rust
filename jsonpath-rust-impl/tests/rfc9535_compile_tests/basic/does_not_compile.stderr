error: unexpected token
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:16:41
   |
16 |     ::jsonpath_rust_impl::json_query!($.&);
   |                                         ^

error: unexpected token
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:22:41
   |
22 |     ::jsonpath_rust_impl::json_query!($.1);
   |                                         ^

error: unexpected token, expected `]`
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:28:43
   |
28 |     ::jsonpath_rust_impl::json_query!($[0 2]);
   |                                           ^

error: unexpected token
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:34:41
   |
34 |     ::jsonpath_rust_impl::json_query!($[,0]);
   |                                         ^

error: unexpected end of input
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:40:43
   |
40 |     ::jsonpath_rust_impl::json_query!($[0,]);
   |                                           ^

error: unexpected end of input
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:46:41
   |
46 |     ::jsonpath_rust_impl::json_query!($[]);
   |                                         ^

error: unexpected end of input
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:52:5
   |
52 |     ::jsonpath_rust_impl::json_query!($..);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `::jsonpath_rust_impl::json_query` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected token
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:58:41
   |
58 |     ::jsonpath_rust_impl::json_query!($[@.a]);
   |                                         ^

error: unexpected token
  --> tests/rfc9535_compile_tests/basic/does_not_compile.rs:64:41
   |
64 |     ::jsonpath_rust_impl::json_query!($[$.a]);
   |                                         ^
